cpu_scheduling_algorithms/round_robin.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file
+ * @brief Implementation of Round CPU scheduling algorithm
+ * @details
+ * Round Robin is a CPU scheduling algorithm where each process is assigned a
+ * fixed time slot in a cyclic way. It is basically the preemptive version of
+ * First come First Serve CPU Scheduling algorithm.
+ * @link https://en.wikipedia.org/wiki/Round-robin_scheduling
+ * @author [Varad Pundlik](https://github.com/varadpundlik)
+ *
+ */
+
+#include <algorithm>  // for max function
+#include <cassert>    // fot testing purpose
+#include <iostream>   //for printing results
+#include <queue>      //for job queue
+#include <string>     // for job name
+#include <utility>    //for using pair
+#include <vector>     // for tasklist of processes
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::max;
+using std::pair;
+using std::queue;
+using std::string;
+using std::vector;
+
+/** @class Job
+    @brief class to represent each process or job
+*/
+class Job {
+    int id;
+    string name;
+    int at;   // Arrival time
+    int wt{0};  // Waiting time
+    int bt;   // Burst time
+    int b1;   // Remaining burst time
+    int tat{0};  // Turnaround time
+    int ct{0};   // Completion time
+
+ public:
+    Job(int i, string n, int a, int b)
+        : id(i), name(std::move(std::move(n))), at(a), bt(b), b1(bt) {}
+    /** @brief prints the process data
+     *  @returns void
+     */
+    void printJob() {
+        cout << name << endl;
+        cout << "id : " << id << endl;
+        cout << "waiting time : " << wt << endl;
+        cout << "burst time : " << bt << endl;
+        cout << "turn around time : " << tat << endl;
+        cout << "completion time : " << ct << endl;
+    }
+
+    friend class Scheduler;
+};
+/** @class Schedduler
+ *  @brief class to performs schedulig algorithms
+ *  @param taskList List of processes
+ *  @param num number of jobs
+ */
+class Scheduler {
+    int num;               // Number of jobs
+    vector<Job> taskList;  // List of jobs
+    double avgWT{0};       // Average waiting time
+    double avgTAT{0};      // Average turnaround time
```

Comment:
```suggestion
    int num;               ///< Number of jobs
    vector<Job> taskList;  ///< List of jobs
    double avgWT{0};       ///< Average waiting time
    double avgTAT{0};      ///< Average turnaround time
```

as mentioned above, variable names with more meaning

Reply:
updated variable names in new commit